pkg/assembler/backends/ent/backend/package.go

```diff
@@ -368,13 +363,9 @@ func packageNameQuery(spec *model.PkgSpec) predicate.PackageName {
 	}
 	query := []predicate.PackageName{
 		optionalPredicate(spec.ID, IDEQ),
-		optionalPredicate(spec.Name, packagename.Name),
-		packagename.HasNamespaceWith(
-			optionalPredicate(spec.Namespace, packagenamespace.Namespace),
-			packagenamespace.HasPackageWith(
-				optionalPredicate(spec.Type, packagetype.Type),
-			),
-		),
+		optionalPredicate(spec.Name, packagename.NameEQ),
+		optionalPredicate(spec.Namespace, packagename.NamespaceEQ),
+		optionalPredicate(spec.Namespace, packagename.TypeEQ),
```

Comment:
```suggestion
		optionalPredicate(spec.Type, packagename.TypeEQ),
```

Reply:
oh good catch. Fixed it as I address the other comments